### PR TITLE
FDS-71, 200 "Sending information to Restaurant and Food delivery services via gRPC"

### PR DESCRIPTION
**Pull request by TonySinero**

### CODE REVIEW

**Comment by TimofeiShorokhov** [comment]
approve

File: GRPC/courierServer.proto [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+message ServicesResponse {
+  repeated DeliveryService services = 1;
+}
+
+message DeliveryService {
+  int64  Id = 1;
+  string ServiceName = 2;
+  string ServiceEmail = 3;
+  string ServicePhoto = 4;
```
Comment (Ann40a) [inline GRPC/courierServer.proto]: you don't need service prefix here

File: controller/CouriersController.go [hunk 2]
```diff
@@ -77,3 +77,31 @@ func (h *Handler) SaveCourier(ctx *gin.Context) {
...
+// @Description put courier status by courier ID
+// @Tags Courier
+// @Accept  json
+// @Produce  json
+// @Param id path int true "ID"
+// @Success 200 {object} int
+// @Failure 400 {string} string
+// @Failure 500 {string} err
+// @Router /courier/{id} [put]
+func (h *Handler) ChangeCourierStatus(ctx *gin.Context) {
```
Comment (Ann40a) [inline controller/CouriersController.go]: If route implies updating entire courier, why controller name says only about status?

File: controller/CouriersController.go [hunk 3]
```diff
@@ -77,3 +77,31 @@ func (h *Handler) SaveCourier(ctx *gin.Context) {
...
+	idQuery := ctx.Param("id")
+	id, err := strconv.Atoi(idQuery)
+	if err != nil {
+		ctx.JSON(http.StatusBadRequest, gin.H{"Error with query parameter": err})
+		return
+	}
+	courierId, err := h.services.CourierApp.ChangeCourierStatus(uint16(id))
+	if err != nil {
+		ctx.JSON(http.StatusInternalServerError, gin.H{"No such order": err})
```
Comment (Ann40a) [inline controller/CouriersController.go]: so if it is not found then it should be appropriate status, if it is server error then different message.

File: dao/OrderRepository.go [hunk 4]
```diff
@@ -238,3 +240,40 @@ func (r *OrderPostgres) GetDetailedOrderByIdFromDB(Id int) (*DetailedOrder, erro
...
+		log.Fatalf("CreateOrder:%s", err)
+		return &emptypb.Empty{}, fmt.Errorf("CreateOrder:%w", err)
+	}
+	return &emptypb.Empty{}, nil
+}
+
+func (r *OrderPostgres) GetServices(in *emptypb.Empty) (*courierProto.ServicesResponse, error) {
+	var Services courierProto.ServicesResponse
+
+	transaction, err := r.db.Begin()
```
Comment (Ann40a) [inline dao/OrderRepository.go]: you don't need transaction when you do just single select

**Review by Ann40a — APPROVED** [review]

**Pull request merged**